### ISSUE TITLE
Failed to start a new language worker for runtime: dotnet-isolated

### ISSUE THREAD

**Issue by lommez** [Azure/azure-functions-dotnet-worker#1514]
Hi, 

My function SOMETIMES throws a TimeoutException and the error message is: Failed to start a new language worker for runtime: dotnet-isolated

Here is the stacktrace

```
2023-05-05T03:13:53.784 [Error] Failed to start a new language worker for runtime: dotnet-isolated.
System.TimeoutException : The operation has timed out.
   at Microsoft.Azure.WebJobs.Script.Grpc.GrpcWorkerChannel.PendingItem.OnTimeout() at /_/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs : 1455
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Script.Grpc.GrpcWorkerChannel.StartWorkerProcessAsync(CancellationToken cancellationToken) at /_/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs : 275
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Script.Workers.Rpc.RpcFunctionInvocationDispatcher.InitializeJobhostLanguageWorkerChannelAsync(??) at /_/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs : 152
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Script.Workers.Rpc.RpcFunctionInvocationDispatcher.InitializeJobhostLanguageWorkerChannelAsync(??) at /_/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs : 144
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Script.Workers.Rpc.RpcFunctionInvocationDispatcher.InitializeJobhostLanguageWorkerChannelAsync(??) at /_/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs : 135
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Script.Workers.Rpc.RpcFunctionInvocationDispatcher.<>c__DisplayClass55_0.<StartWorkerProcesses>b__0(??) at /_/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs : 215
```

And here is the packages versions that I'm using:

![image](https://user-images.githubusercontent.com/10900667/236489913-a7d0dd1e-aa0a-420d-b8a6-2dd0dec5cc02.png)

What can cause that intermittent exception ?

**Comment by kshyju** [comment]
Hi @lommez, We recently identified a race condition issue which might cause this behavior and fixed it in #1508 and released in worker package [1.14.0](https://github.com/Azure/azure-functions-dotnet-worker/releases/tag/1.14.0). Please try this version.

https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker/1.14.0

**Comment by lommez** [comment]
@kshyju I've updated my application and those exceptions are gone!!!
Thank you so much!!!
